src/transformers/modeling_utils.py

```diff
@@ -2162,7 +2162,7 @@ def find_submodule_and_param_name(model, long_key):
             state_dict = torch.load(archive_file, map_location="cpu")
 
             # materialize state_dict entries one by one on CPU
-            for k in loaded_state_dict_keys:
+            for k in state_dict.keys():
```

Comment:
What's the reason this was changed? It was a bug right?

cc @sgugger

Reply:
yes, with multiple checkpoints you can't find all the keys of the model in every checkpoints, because they were splitted

Reply:
and it sounds like we are missing some tests.

Reply:
If this is not addressed here, loading a big model in 1x memory consumption from chucked checkpoints is not possible. I think my approach is safer and will fail in the cases that are supposed to fail, otherwise, we are silently skipping keys that may be supposed to be there. Let me know your thoughts.

Reply:
As said before:
> Any fix for low memory loading and sharded checkpoints would need to be in its own PR with a test.

PRs should be focused on one task: adding a new model is different from fixing a bug. :-)

Reply:
>otherwise, we are silently skipping keys that may be supposed to be there. Let me know your thoughts.

That was exactly my point in reverse. What if we need to skip some keys? 

My assumption is that `loaded_state_dict_keys` is 100% correct because the model was initialized using those keys. So it's the master list of what we expect to load.

and if we are missing some keys, we should have sorted it out earlier when we populated `loaded_state_dict_keys`

Reply:
I think what Sylvain is suggesting is that perhaps keep the fix in here, do a separate PR that is just the fix, merge it and then rebase this PR to the merged fix - does it make sense?

Reply:
Two of the models added need the fix, I thought it was not a bad idea to catch two birds with one stone. Gonna revert it

Reply:
so that's what I'm proposing - don't revert anything - add another PR with just the fix - and then when it gets merged - rebased both models - meanwhile you can continue working

Reply:
Thanks, [opened a new PR](https://github.com/huggingface/transformers/pull/16643)

Reply:
Thanks to everybody, I've updated this file to match the one in `main`